testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -542,6 +548,102 @@ public void testOrcTableCreatedInTrino()
+        StorageFormatsHiveCompatibilityColumnData[] storageFormatsHiveCompatibilityTableData = new StorageFormatsHiveCompatibilityColumnData[] {
+                new StorageFormatsHiveCompatibilityColumnData("c_boolean", "BOOLEAN", "true", true),
+                new StorageFormatsHiveCompatibilityColumnData("c_tinyint", "TINYINT", "127", 127),
+                new StorageFormatsHiveCompatibilityColumnData("c_smallint", "SMALLINT", "32767", 32767),
+                new StorageFormatsHiveCompatibilityColumnData("c_int", "INTEGER", "2147483647", 2147483647),
+                new StorageFormatsHiveCompatibilityColumnData("c_bigint", "BIGINT", "9223372036854775807", 9223372036854775807L),
+                new StorageFormatsHiveCompatibilityColumnData("c_real", "REAL", "123.345", 123.345d),
+                new StorageFormatsHiveCompatibilityColumnData("c_double", "DOUBLE", "234.567", 234.567d),
+                new StorageFormatsHiveCompatibilityColumnData("c_decimal", "DECIMAL(10,0)", "346", new BigDecimal("346")),
+                new StorageFormatsHiveCompatibilityColumnData("c_decimal_w_params", "DECIMAL(10,5)", "345.67800", new BigDecimal("345.678")),
+                new StorageFormatsHiveCompatibilityColumnData("c_char", "CHAR(10)", "'ala ma    '", "ala ma    "),
+                new StorageFormatsHiveCompatibilityColumnData("c_varchar", "VARCHAR(10)", "'ala ma kot'", "ala ma kot"),
+                new StorageFormatsHiveCompatibilityColumnData("c_string", "VARCHAR", "'ala ma kota'", "ala ma kota"),
+                new StorageFormatsHiveCompatibilityColumnData("c_binary", "VARBINARY", "X'62696e61727920636f6e74656e74'", "binary content".getBytes(StandardCharsets.UTF_8)),
+                new StorageFormatsHiveCompatibilityColumnData("c_date", "DATE", "DATE '2015-05-10'", Date.valueOf(LocalDate.of(2015, 5, 10))),
+                new StorageFormatsHiveCompatibilityColumnData("c_timestamp", "TIMESTAMP", "TIMESTAMP '2015-5-10 12:15:35.123'",
+                        isAvroStorageFormat && isHiveWithBrokenAvroTimestamps()
+                                // TODO (https://github.com/trinodb/trino/issues/1218) requires https://issues.apache.org/jira/browse/HIVE-21002
+                                ? Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 6, 30, 35, 123_000_000))
+                                : Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 12, 15, 35, 123_000_000))),
+                new StorageFormatsHiveCompatibilityColumnData("c_array", "ARRAY(INTEGER)", "ARRAY[1, 2, 3]", "[1,2,3]"),
+                new StorageFormatsHiveCompatibilityColumnData("c_map", "MAP(VARCHAR, VARCHAR)", "MAP(ARRAY['foo'], ARRAY['bar'])", "{\"foo\":\"bar\"}"),
+                new StorageFormatsHiveCompatibilityColumnData("c_row", "ROW(f1 INTEGER, f2 VARCHAR)", "ROW(42, 'Trino')", "{f1:42,f2:\"Trino\"}")
+        };
+
+        Function<StorageFormatsHiveCompatibilityColumnData, Boolean> storageFormatSupportedColumnTypePredicate = data -> !(
```

Comment:
This filters out test cases. That's fine, however, it could be made simpler

```
testColumnData = new ArrayList<>();
....
if (isParquetStorageFormat && isHiveVersionBefore12() && "DATE".equals(data.trinoColumnType)) {
  // The PARQUET storage format does not support DATE type in CDH5 distribution
} 
else  {
   testCases.add( ... date related test cases );
}
```

I know it's not as generic, due to how conditions are structured, but feels simpler to realize the test case is included conditionally.

Reply:
This is a very good point. I've refactored the code and it seems indeed much more straightforward to read and maintain it. Thank you for bringing this up.